torchgeo/datasets/cloud_cover.py

```diff
@@ -0,0 +1,410 @@
+            sample = self.transforms(sample)
+
+        return sample
+
+    def _load_image(self, index: int) -> Tensor:
+        """Load all source images for a chip.
+
+        Args:
+            index: position of the indexed chip
+
+        Returns:
+            a tensor of stacked source image data
+        """
+        source_asset_paths = self.chip_paths[index]["source"]
+        images = []
+        for path in source_asset_paths:
+            with rasterio.open(path) as image_data:
+                image_array = image_data.read(1).astype(np.int32)
+                images.append(image_array)
+        image_stack: "np.typing.NDArray[np.int_]" = np.stack(images, axis=0)
+        image_tensor = torch.from_numpy(image_stack)
+        return image_tensor
+
+    def _load_target(self, index: int) -> Tensor:
+        """Load label image for a chip.
+
+        Args:
+            index: position of the indexed chip
+
+        Returns:
+            a tensor of the label image data
+        """
+        label_asset_path = self.chip_paths[index]["target"][0]
+        with rasterio.open(label_asset_path) as target_data:
+            target_img = target_data.read(1).astype(np.int32)
+
+        target_array: "np.typing.NDArray[np.int_]" = np.array(target_img)
+        target_tensor = torch.from_numpy(target_array)
+        return target_tensor
+
+    @staticmethod
+    def _validate_stac_object(object_path: str) -> bool:
+        """Validates a STAC object JSON filepath exists.
+
+        Returns:
+            True if the object is a valid JSON file
+
+        Raises:
+            RuntimeError: if JSON or TIF file is not a valid object on disk
+        """
+        if not os.path.exists(object_path):
+            raise RuntimeError(
+                f"{object_path} not a valid STAC object path in the Catalog. "
+                + "Make sure all files uncompressed completely."
+            )
+        return True
+
+    @staticmethod
+    def _read_json_data(object_path: str) -> Any:
+        with open(object_path) as read_contents:
+            json_data = json.load(read_contents)
+        return json_data
+
+    def _load_items(self, item_json: str) -> Dict[str, List[str]]:
+        """Loads the label item and corresponding source items.
+
+        Args:
+            item_json: a string path to the item JSON file on disk
+
+        Returns:
+            a dictionary with paths to the source and target TIF filenames
+        """
+        item_meta = {}
+
+        if self._validate_stac_object(item_json):
+            label_data = self._read_json_data(item_json)
+            label_asset_path = os.path.join(
+                os.path.split(item_json)[0], label_data["assets"]["labels"]["href"]
+            )
+            item_meta["target"] = [label_asset_path]
+
+            if self._validate_stac_object(label_asset_path):
+                source_item_hrefs = [
+                    os.path.join(self.root, link["href"].replace("../../", ""))
```

Comment:
Looks like the cleanest way to do this is `os.path.abspath`. I will push that change in a commit.

Reply:
Update on this, I actually ran into issues using `os.path.abspath`, `os.path.realpath` and `Path().absolute()`. None of these build the appropriate file paths because the path provided in the `stac.json` file for a STAC Item is relative to the collection, e.g.:

> ../../ref_cloud_cover_detection_challenge_v1_test_source/ref_cloud_cover_detection_challenge_v1_test_source_aaaa/stac.json

I'm not sure what the best way to do this is without calling `.replace('../../', '')`